File: recursion/1878-check-if-array-is-sorted-and-rotated/check-if-array-is-sorted-and-rotated.java
class Solution {
    public boolean check(int[] nums) {
        int min = 0;
        for(int i = 0 ; i<nums.length-1;i++){
            if(nums[i]>nums[i+1]){
                min = i+1;
            }
        }
        int count = 0;
        int k = 0;
        int cur = 0;
        int nex = 0;
        while(count<nums.length-1){
            cur = (min+k)%nums.length;
            nex = (min+k+1)%nums.length;
            if(nums[nex]>=nums[cur]){
                count++;
                k++;
                continue;
            }else{
                return false;
            }
        }
        return true;
    }
}

File: recursion/1878-check-if-array-is-sorted-and-rotated/check-if-array-is-sorted-and-rotated.cpp
class Solution {
public:
    bool check(vector<int>& nums) {
        int count = 0;
        int size = nums.size();
        
        for(int i = 0 ; i < size ; i ++){
            if(count >1)
                break;
            if(nums[i] > nums[(i+1)%size ] ){
                count++;
            }else{
                continue;
            }
        }
        if(count<=1){
            return true;
        }else{
            return false;
        }
    }
};